WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooAddCustomPackageView.swift

```diff
@@ -42,113 +43,142 @@ struct WooAddCustomPackageView: View {
         .roundedBorder(cornerRadius: 8, lineColor: Color(.separator), lineWidth: 1)
     }
 
+    private var loadingView: some View {
+        VStack {
+            Spacer()
+            HStack {
+                Spacer()
+                if customPackageViewModel.isLoadingStoreOptions {
+                    ActivityIndicator(isAnimating: .constant(true), style: .large)
+                }
+                else {
+                    Button {
+                        customPackageViewModel.loadStoreOptions()
+                    } label: {
+                        Image(systemName: "arrow.trianglehead.counterclockwise")
+                    }
+                }
+                Spacer()
+            }
+            Spacer()
+        }
+    }
+
     var body: some View {
         GeometryReader { geometry in
-            ScrollViewReader { proxy in
-                ScrollView {
-                    VStack(alignment: .leading, spacing: Constants.defaultVerticalSpacing) {
-                        HStack {
-                            Text(Localization.packageType)
-                                .font(.subheadline)
-                            Spacer()
-                        }
-                        packageTypeSelectionView
-                        VStack {
-                            AdaptiveStack(spacing: 8) {
-                                ForEach(WooShippingPackageUnitType.dimensionUnits, id: \.self) { dimensionUnit in
-                                    unitInputView(for: dimensionUnit, unit: customPackageViewModel.dimensionsUnit)
+            if let storeOptions = customPackageViewModel.storeOptions {
+                ScrollViewReader { proxy in
+                    ScrollView {
+                        VStack(alignment: .leading, spacing: Constants.defaultVerticalSpacing) {
+                            HStack {
+                                Text(Localization.packageType)
+                                    .font(.subheadline)
+                                Spacer()
+                            }
```

Comment:
This part looks like a lot of code change but basically if/else is added to show the content if the store options are loaded and show loading view if not.